ext/native/thin3d/thin3d_vulkan.cpp

```diff
@@ -1470,585 +1237,97 @@ uint32_t VKContext::GetDataFormatSupport(DataFormat fmt) const {
 }
 
 // color must be 0, for now.
 void VKContext::BindFramebufferAsTexture(Framebuffer *fbo, int binding, FBChannel channelBit, int attachment) {
 	VKFramebuffer *fb = (VKFramebuffer *)fbo;
-	boundImageView_[0] = fb->color.view;
-	// If we already have the right layout, nothing else to do.
-	if (fb->color.layout == VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL)
-		return;
-
-	VkCommandBuffer transitionCmdBuf;
-	if (fb->cmdBuf && fb->frameCount == frameNum_) {
-		// If the framebuffer has a "live" command buffer, we can directly use it to transition it for sampling.
-		transitionCmdBuf = fb->cmdBuf;
-	} else {
-		// If not, we can just do it at the "start" of the frame.
-		transitionCmdBuf = vulkan_->GetInitCommandBuffer();
-	}
 
-	VkAccessFlags srcAccessMask;
-	VkPipelineStageFlags srcStage;
-	switch (fb->color.layout) {
-	case VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL:
-		srcAccessMask = VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT | VK_ACCESS_COLOR_ATTACHMENT_READ_BIT;
-		srcStage = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
-		break;
-	case VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL:
-		srcAccessMask = VK_ACCESS_TRANSFER_WRITE_BIT;
-		srcStage = VK_PIPELINE_STAGE_TRANSFER_BIT;
-		break;
-	}
-	TransitionImageLayout2(transitionCmdBuf, fb->color.image, VK_IMAGE_ASPECT_COLOR_BIT,
-		fb->color.layout, VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL,
-		srcStage, VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT,
-		srcAccessMask, VK_ACCESS_SHADER_READ_BIT);
-	fb->color.layout = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
+	int aspect = 0;
+	if (channelBit & FBChannel::FB_COLOR_BIT) aspect |= VK_IMAGE_ASPECT_COLOR_BIT;
+	if (channelBit & FBChannel::FB_DEPTH_BIT) aspect |= VK_IMAGE_ASPECT_DEPTH_BIT;
+	if (channelBit & FBChannel::FB_STENCIL_BIT) aspect |= VK_IMAGE_ASPECT_STENCIL_BIT;
+	boundImageView_[0] = renderManager_.BindFramebufferAsTexture(fb->GetFB(), binding, aspect, attachment);
 }
 
 uintptr_t VKContext::GetFramebufferAPITexture(Framebuffer *fbo, int channelBit, int attachment) {
-	// TODO: Insert transition at the end of the previous command buffer, or the one that rendered to it last.
-	VKFramebuffer *fb = (VKFramebuffer *)fbo;
-	return (uintptr_t)fb->color.image;
+	return 0;
 }
 
 void VKContext::GetFramebufferDimensions(Framebuffer *fbo, int *w, int *h) {
 	VKFramebuffer *fb = (VKFramebuffer *)fbo;
 	if (fb) {
-		*w = fb->width;
-		*h = fb->height;
+		*w = fb->GetFB()->width;
+		*h = fb->GetFB()->height;
 	} else {
-		*w = vulkan_->GetWidth();
-		*h = vulkan_->GetHeight();
+		*w = vulkan_->GetBackbufferWidth();
+		*h = vulkan_->GetBackbufferHeight();
 	}
 }
 
 void VKContext::HandleEvent(Event ev, int width, int height, void *param1, void *param2) {
+	switch (ev) {
+	case Event::LOST_BACKBUFFER:
+		renderManager_.DestroyBackbuffers();
+		break;
+	case Event::GOT_BACKBUFFER:
+		renderManager_.CreateBackbuffers();
+		break;
+	default:
+		break;
```

Comment:
Hmm, why not an assert?  I'd rather be explicit, it's not uncommon we'd add an event and forget to update D3D11 or somewhere.

-[Unknown]